### PR TITLE
AISOTT: Refactor MBID mapping tables to allow easy rechecking on unmatched data

### PR DESCRIPTION
**Pull request by mayhem**
This PR renames the two mapping tables to be more clear and changes the FK relationship between the two tables. This change simplifies the ability to re-check items that were previously not matched. There is the beginnings of automatic rechecking, but I decided to finish that work later once we've manually rechecked listens for 2021 so that we can make better year end reports.

In order to recheck an item in the mapping, set the last_updated timestamp to 1970-01-01 and it will be checked during the next batch of legacy/recheck.

NOTE: Do not merge this yet -- it needs to be carefully released.

### CODE REVIEW

File: listenbrainz/db/dump.py [hunk 1]
```diff
@@ -111,21 +111,21 @@
 }
 
 PUBLIC_TABLES_TIMESCALE_DUMP = {
-    'listen_mbid_mapping': (
-        'id',
+    'mbid_mapping_metadata': (
```
Comment (amCap1712) [inline listenbrainz/db/dump.py:114]: Update schema version also

File: listenbrainz/db/tests/test_feedback.py [hunk 2]
```diff
@@ -58,32 +58,36 @@ def insert_test_data_with_metadata(self, user_id, neg_score=False):
         """ Insert test data with metadata into the database """
 
         with msb_db.engine.connect() as connection:
-            msid = submit_recording(connection, self.sample_recording)
+            msid = get_id_from_meta_hash(connection, self.sample_recording)
```
Comment (amCap1712) [inline listenbrainz/db/tests/test_feedback.py:61]: Why is this needed especially why in test only?

File: listenbrainz/mbid_mapping_writer/job_queue.py [hunk 3]
```diff
@@ -86,11 +108,60 @@ def load_legacy_listens(self):
...
+                count += 1
+
+        return count
+
     def add_legacy_listens_to_queue(self):
         """Fetch more legacy listens from the listens table by doing an left join
            on the matched listens, finding the next chunk of legacy listens to look up.
            Listens are added to the queue with a low priority."""
 
+        legacy_query = """SELECT data->'track_metadata'->'additional_info'->>'recording_msid'::TEXT AS recording_msid
```
Comment (amCap1712) [inline listenbrainz/mbid_mapping_writer/job_queue.py:152]: This query seemed weird at first look.  Maybe a comment to tell its to select not yet mapped msids? Also, would it make to use a WHERE EXISTS or WHERE IN?

**Review by amCap1712 — APPROVED** [review]
Other than the comments, LGTM.

File: listenbrainz/db/tests/test_feedback.py (hunk 2 above)
Comment (mayhem) [reply listenbrainz/db/tests/test_feedback.py:61]: Running tests repeatedly (using test -u) causes this test to fail with a duplicate row violation. Checking to see if the data is already in the DB prevent this error.

File: listenbrainz/db/dump.py (hunk 1 above)
Comment (mayhem) [reply listenbrainz/db/dump.py:114]: Updated both of them now.

File: listenbrainz/mbid_mapping_writer/job_queue.py (hunk 3 above)
Comment (mayhem) [reply listenbrainz/mbid_mapping_writer/job_queue.py:152]: :+1:

File: listenbrainz/db/tests/test_feedback.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz/db/tests/test_feedback.py:61]: Oh, weird. because submit_recording is supposed to take care of that. Anyways fine for a test. We can investigate later.
